src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -2062,25 +2063,47 @@ ContextCoversJoinRestriction(JoinRestrictionContext *joinRestrictionContext,
 			continue;
 		}
 
+		List *joinRestrictInfoListInTest =
+			joinRestrictionInTest->joinRestrictInfoList;
+		bool hasJoinRestriction = list_length(joinRestrictInfoListInTest) == 0;
```

Comment:
this should be `> 0` and the if statement should use `!hasJoinRestriction`

Reply:
oh this is true, I will apply that